packages/terminal/src/browser/base/terminal-widget.ts

```diff
@@ -58,6 +59,9 @@ export abstract class TerminalWidget extends BaseWidget {
      */
     abstract processInfo: Promise<TerminalProcessInfo>;
 
+    /** The extensions contributing to the environment of this terminal */
+    abstract contributingExtensions: Promise<Map<string, string | MarkdownString | undefined>>;
```

Comment:
I think the name is quite non-descriptive here: what is being held in this map?

Reply:
Went with `envVarCollectionDescriptionsByExtension`